draft-ietf-tls-dtls13.md

```diff
@@ -750,6 +765,55 @@ because an attacker will have difficulty forging a datagram that will
 not be rejected by the transport layer.
 
 
+### AEAD Limits
+
+Section 5.5 of TLS {{!TLS13}} defines limits on the number of records that can
+be protected using the same keys. These limits are specific to an AEAD
+algorithm, and apply equally to DTLS. Implementations SHOULD NOT protect more
+records than allowed by the limit specified for the negotiated AEAD.
+
+{{!TLS13}} does not specify a limit for AEAD_AES_128_CCM, but the analysis in
+{{ccm-bounds}} shows that a limit of 2^23 packets can be used to obtain the
+same confidentiality protection as the limits specified in TLS.
+
+The usage limits defined in TLS 1.3 exist for protection against attacks
+on confidentiality and apply to successful applications of AEAD protection. The
+integrity protections in authenticated encryption also depend on limiting the
+number of attempts to forge packets. TLS achieves this by closing connections
+after any record fails an authentication check. In comparison, DTLS ignores any
+packet that cannot be authenticated, allowing multiple forgery attempts.
+
+Implementations MUST count the number of received packets that fail
+authentication. If the number of packets that fail authentication exceeds a
+limit that is specific to the AEAD in use, an implementation MUST immediately
+close the connection. Implementations SHOULD initiate a key update before
+reaching this limit. Applying a limit reduces the probability that an attacker
+is able to successfully forge a packet; see {{AEBounds}} and {{ROBUST}}.
```

Comment:
A few points:

1. You need to initiate a key update with UpdateRequested, right?
2. The limit is actually per key, right?
3. Once the update has started, you could presumably drop the new epoch keys early rather than terminating the connection.

Reply:
1. Good point.  QUIC doesn't have this distinction, but it matters here.

2. Yes, per key.  Will clarify (and in QUIC too).

3. That's a good point.  I'll add something.